### PR TITLE
MRG: Fewer warnings during install

### LINKED ISSUE #5390: installation problems on python3
EDIT: To do:

- [ ] update install docs to mention warnings are not fatal
- [ ] update `environment.yml` to avoid mkl warnings
- [ ] update install docs to mention the MESA version workaround @nugenta found

Original post:

I had this working, then a system upgrade broke my conda environments.  Now I'm running a miniconda install. I've tried:

$ curl -O https://raw.githubusercontent.com/mne-tools/mne-python/master/environment.yml
$ conda env create -f environment.yml
$ source activate mne
$ mne coreg

I've pasted in the voluminous error at the bottom. The interface pops up, and looks fine, but no brain is shown and the error pasted in below is resent everytime you touch anything. 

I've tried specifying ETS_TOOKIT and QT_API and then I get 
"RuntimeError: No pyface.toolkits plugin found for toolkit qt5' 

I tried updating pyface; that downgrades twister.

I tried updating qt to 5.10, also no luck.

I did notice a potential problem:
$ conda list | grep vtk
vtk                       8.1.1                     <pip>
vtk                       8.1.0           py36h9686630_201  

I don't know if the two versions of vtk could be conflicting.

ERROR: In /work/standalone-x64-build/VTK-source/Rendering/OpenGL2/vtkShaderProgram.cxx, line 431
vtkShaderProgram (0x556abe29b9b0): 

ERROR: In /work/standalone-x64-build/VTK-source/Rendering/OpenGL2/vtkShaderProgram.cxx, line 430
vtkShaderProgram (0x556abe29c3a0): 1: #version 120
2: #ifdef GL_ES
3: #if __VERSION__ == 300
4: #define attribute in
5: #define varying out
6: #endif // 300
7: #else // GL_ES
8: #define highp
9: #define mediump
10: #define lowp
11: #if __VERSION__ == 150
12: #define attribute in
13: #define varying out
14: #endif
15: #endif // GL_ES
16: 
17: 
18: /*=========================================================================
19: 
20:   Program:   Visualization Toolkit
21:   Module:    vtkPolyDataVS.glsl
22: 
23:   Copyright (c) Ken Martin, Will Schroeder, Bill Lorensen
24:   All rights reserved.
25:   See Copyright.txt or http://www.kitware.com/Copyright.htm for details.
26: 
27:      This software is distributed WITHOUT ANY WARRANTY; without even
28:      the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
29:      PURPOSE.  See the above copyright notice for more information.
30: 
31: =========================================================================*/
32: 
33: attribute vec4 vertexMC;
34: 
35: // frag position in VC
36: varying vec4 vertexVCVSOutput;
37: 
38: // optional normal declaration
39: attribute vec3 normalMC;
40: uniform mat3 normalMatrix;
41: varying vec3 normalVCVSOutput;
42: 
43: // extra lighting parameters
44: //VTK::Light::Dec
45: 
46: // Texture coordinates
47: //VTK::TCoord::Dec
48: 
49: // material property values
50: //VTK::Color::Dec
51: 
52: // clipping plane vars
53: //VTK::Clip::Dec
54: 
55: // camera and actor matrix values
56: uniform mat4 MCDCMatrix;
57: uniform mat4 MCVCMatrix;
58: 
59: // Apple Bug
60: //VTK::PrimID::Dec
61: 
62: // Value raster
63: //VTK::ValuePass::Dec
64: 
65: void main()
66: {
67:   //VTK::Color::Impl
68: 
69:   normalVCVSOutput = normalMatrix * normalMC;
70: 
71:   //VTK::TCoord::Impl
72: 
73:   //VTK::Clip::Impl
74: 
75:   //VTK::PrimID::Impl
76: 
77:   vertexVCVSOutput = MCVCMatrix * vertexMC;
78:   gl_Position = MCDCMatrix * vertexMC;
79: 
80: 
81:   //VTK::ValuePass::Impl
82: 
83:   //VTK::Light::Impl
84: }
85: 

ERROR: In /work/standalone-x64-build/VTK-source/Rendering/OpenGL2/vtkShaderProgram.cxx, line 431
vtkShaderProgram (0x556abe29c3a0):

### PR DESCRIPTION
**Pull request by larsoner**
1. Multiple users have complained about the Cython warnings. This should get rid of those.
2. I don't think we need to have the doc-building deps in there by default, since that's really a dev thing, and devs can easily install those deps. (Unit test stuff is still nice for users who submit bug reports.)
3. I think we can do PySurfer on standard `pip` instead of from `master` nowadays.
4. There were some tweaks and tips proposed in #5390 that are included here.
5. We have `Issue` templates on GitHub now, so no need to duplicate instructions in our own docs.

Closes #5390.

### CODE REVIEW

**Comment by larsoner** [comment]
Updated install docs:

https://9056-1301584-gh.circle-artifacts.com/0/html/install_mne_python.html

Updated advanced docs:

https://9056-1301584-gh.circle-artifacts.com/0/html/advanced_setup.html#off-screen-rendering-on-linux-with-mesa

I don't like that it pushes our install docs longer, but it's probably worth it so users can feel comfortable after setting up their env.

File: doc/contributing.rst [hunk 1]
```diff
@@ -121,8 +121,8 @@ follows the `NumPy docstring standard`_.
 
 Documentation is automatically built remotely during pull requests. If
 you want to also test documentation locally, you will need to install
-``sphinx sphinx-gallery sphinx_bootstrap_theme numpydoc``, and then within
-the ``mne/doc`` directory do:
+``sphinx sphinx-gallery sphinx_bootstrap_theme sphinx_fontawesome``, and then
+within the ``mne/doc`` directory do:
```
Comment (agramfort) [inline doc/contributing.rst]: we do need also numpydoc.

File: environment.yml [hunk 2]
```diff
@@ -18,7 +19,6 @@ dependencies:
 - jupyter
 - pytest
 - pytest-cov
-- sphinx
 - joblib
 - psutil
 - numpydoc
```
Comment (agramfort) [inline environment.yml]: why keep numpydoc? you decided to remove all packages related to doc building?

File: doc/contributing.rst (hunk 1 above)
Comment (larsoner) [reply doc/contributing.rst]: it's in `environment.yml`, no need to have it here (I'm only listing things not in the env)

File: environment.yml (hunk 2 above)
Comment (larsoner) [reply environment.yml]: we use it in `test_docstring_parameters.py`

**Comment by larsoner** [comment]
Thanks @nugenta I added a little note about `llvmpipe` possibly outperforming `swr`

**Comment by agramfort** [comment]
@larsoner merge if you're happy

**Pull request merged**